### PR TITLE
[Resources and support] Prevent some QA from becoming standalone pages

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
Some nodes of the type `q_a` should generate standalone pages on the site, but some are just used to compose Multiple FAQ pages. In the latter case, we need to exclude them from the build. We check the value of `fieldStandalonePage`  to determine this.

-- edit - this PR has grown to include -

1. Excludes QA pages from the site that are marked as "standalone page" = false
1. Fixes the issue with the spooky ghost accordion button
1. Removes tags from search results
1. Updates search result criteria to unpluralize queries and do a couple other things
1. Updates the Multiple FAQ page to check for fieldAccordionDisplay and not display as an accordion if false
1. Implement some specific requirements for how table of contents are generated on this website section

https://github.com/department-of-veterans-affairs/va.gov-team/issues/14929

## Testing done
Ran locally and observed output

## Screenshots
N/A

## Acceptance criteria
- [ ] QA pages are not build to website if `fieldStandalonePage` is not true

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/constants/content-modeling.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  SUPPORT_RESOURCES_ARTICLE_LISTING: 'support_resources_article_listing',
+  SUPPORT_RESOURCES_DETAIL_PAGE: 'support_resources_detail_page',
+  SUPPORT_SERVICE: 'support_service',
+  VA_FORM: 'va_form',
+  VAMC_OPERATING_STATUS_AND_ALERTS: 'vamc_operating_status_and_alerts',
+};
+
+module.exports = {
+  ENTITY_BUNDLES,
+};
```
Comment (ncksllvn) [inline src/site/constants/content-modeling.js:51]: IDK about this file name but I am a little tired of magic strings floating around to represent entityBundles

Comment (kelsonic) [reply src/site/constants/content-modeling.js:51]: 🧙 lol

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉 Thanks!

**Comment by jenniferlee-dsva** [comment]
Including a screenshot from a sample node where this bug appeared. We can check this page again to validate: http://preview-prod.vfs.va.gov/preview?nodeId=8483

image

File: src/site/constants/drupals.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const DRUPAL_STAGING = {
 };
 
 const DRUPAL_PROD = {
-  address: 'https://prod.cms.va.gov',
+  address: 'http://learningcenter.demo.ci.cms.va.gov',
```
Comment (ncksllvn) [inline src/site/constants/drupals.js:18]: Uh oh, this was pushed by accident. Reverting...

File: src/site/stages/build/plugins/create-resources-and-support-section.js [hunk 3]
```diff
@@ -323,6 +336,7 @@ function createResourcesAndSupportWebsiteSection(buildOptions) {
   }
 
   return files => {
+    excludeQaNodesThatAreNotStandalonePages(files);
```
Comment (ncksllvn) [inline src/site/stages/build/plugins/create-resources-and-support-section.js:339]: Here is where we basically unpublish QA nodes that have a `fieldStandalonePage` of false

File: src/site/layouts/faq_multiple_q_a.drupal.liquid [hunk 4]
```diff
@@ -54,25 +54,52 @@
...
+                          {% assign bundleComponent = "src/site/paragraphs/" | append: fieldAnswer.entity.entityBundle %}
+                          {% assign bundleComponentWithExtension = bundleComponent | append: ".drupal.liquid" %}
+                          {% include {{ bundleComponentWithExtension }} with entity = fieldAnswer.entity %}
+                        </div>
+                      </li>
+                    {% endif %}
                   {% endfor %}
                 </ul>
-              </div>
+              {% endif %}
```
Comment (ncksllvn) [inline src/site/layouts/faq_multiple_q_a.drupal.liquid:102]: This implements a fix where a `fieldQA.entity` of `null` would crash the template and result in a ghost accordion row

File: src/applications/resources-and-support/hooks/useGetSearchResults.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+      const orderedResults = sortBy(filteredArticles, 'title');
+
+      setResults(orderedResults);
+    },
+    [articles, setResults, query, page],
+  );
+
+  return [results];
+}
```
Comment (ncksllvn) [inline src/applications/resources-and-support/hooks/useGetSearchResults.js:48]: I separated out some of the hooks into their own files now that the logic is lengthier. The search here is kinda complex now but is based on a thread on Slack.

File: src/applications/resources-and-support/components/SearchResult.jsx [hunk 6]
```diff
@@ -35,27 +30,6 @@ export default function SearchResult({ article }) {
...
-        </div>
-      )}
-      {article.fieldTags?.entity?.fieldAudienceBeneficiares?.entity?.name && (
-        <div className="vads-u-margin-top--1">
-          <dfn className="vads-u-font-weight--bold">Audience </dfn>
-          <Tag
-            term={article.fieldTags.entity.fieldAudienceBeneficiares.entity}
-          />
-        </div>
-      )}
```
Comment (ncksllvn) [inline src/applications/resources-and-support/components/SearchResult.jsx:58]: Per a thread on Slack, "tags" are not to be included in search results any longer. I was looking at an outdated design file.

**Review by rianfowler — APPROVED** [review]

**Pull request merged**